src/components/services/Rtorrent.vue

```diff
@@ -0,0 +1,154 @@
+  data: () => ({dl: null, ul: null, count: null, error: null}),
+  // Computed properties for the rate labels.
+  computed: {
+    downRate: function() {
+      return displayRate(this.dl);
+    },
+    upRate: function() {
+      return displayRate(this.ul);
+    },
+  },
+  created() {
+    // Set intervals if configured so the rates and/or torrent count
+    // will be updated.
+    const rateInterval = parseInt(this.item.rateInterval, 10) || 0;
+    const torrentInterval = parseInt(this.item.torrentInterval, 10) || 0;
+
+    if (rateInterval > 0) {
+      setInterval(() => this.fetchRates(), rateInterval);
+    }
+
+    if (torrentInterval > 0) {
+      setInterval(() => this.fetchCount(), torrentInterval);
+    }
+    
+    // Fetch the initial values.
+    this.fetchRates();
+    this.fetchCount();
+  },
+  methods: {
+    // Perform two calls to the XML-RPC service and fetch download
+    // and upload rates. Values are saved to the `ul` and `dl`
+    // properties.
+    fetchRates: async function() {
+      try {
+        await this.getRate('throttle.global_up.rate')
+          .then((ul) => this.ul = ul);
+        await this.getRate('throttle.global_down.rate')
+          .then((dl) => this.dl = dl);
+      } catch {
+        this.error = true;
+      }
+    },
+    // Perform a call to the XML-RPC service to fetch the number of
+    // torrents.
+    fetchCount: async function() {
+      this.getCount().catch(() => this.error = true);
+    },
+    // Fetch a numeric value from the XML-RPC service by requesting
+    // the specified method name and parsing the XML. The response
+    // is expected to adhere to the structure of a single numeric
+    // value.
+    getRate: async function(methodName) {
+      return this.getXml(methodName)
+        .then((xml) => parseInt(xml.getElementsByTagName('value')[0].firstChild.textContent, 10));
+    },
+    // Fetch the numer of torrents by requesting the download list
+    // and counting the number of entries therein.
+    getCount: async function() {
+      return this.getXml('download_list')
+        .then((xml) => {
+          const arrayEl = xml.getElementsByTagName('array');
+          this.count = arrayEl ? arrayEl[0].getElementsByTagName('value').length : 0;
+        });
+    },
+    // Perform a call to the XML-RPC service and parse the response
+    // as XML, which is then returned.
+    getXml: async function(methodName) {
+      const headers = {'Content-Type': 'text/xml'};
+
+      if (this.item.username && this.item.password) {
+        headers['Authorization'] = `${this.item.username}:${this.item.password}`;
+      }
+
+      return fetch(`${this.item.xmlrpc.replace(/\/$/, '')}/RPC2`, {
+        method: 'POST',
+        headers,
+        body: `<methodCall><methodName>${methodName}</methodName></methodCall>`
+      })
+      .then((response) => {
+        if (!response.ok) {
+          throw Error(response.statusText);
```

Comment:
you can save you the error handling by using this.fetch ([custom fetch function](https://github.com/bastienwirtz/homer/blob/main/src/mixins/service.js#L15) that handle error detection. By default it does json parsing but you can disable it (last parameter to false).